backend/helper_zero/views/search.py

```diff
@@ -13,24 +15,52 @@ class SearchView(viewsets.ViewSet):
     """
 
     def list(self, request):
-        delivery = request.query_params.get('delivery', False)
-        item_type = request.query_params.get('item_type')
-        lat = request.query_params.get('lat')
-        lon = request.query_params.get('lon')
-        radius = request.query_params.get('radius', 10)
-        zipcode = request.query_params.get('zipcode')
         offset = int(request.query_params.get('offset', 0))
-
-        lat = float(lat) if lat else None
-        lon = float(lon) if lon else None
-        delivery_values = [True] if delivery else [True, False]
-
-        org_search_results = _get_org_search_results(
-            zipcode, lat, lon, radius, delivery_values, offset
-        )
-
-        response = _filter_org_list_by_donation_requests(org_search_results, item_type)
-        return Response(response)
+        item_type = request.query_params.get('item_type')
+        location_params, org_filter_params, donation_request_filter_params = _process_params(request.query_params)
+
+        try:
+            org_search_results = _get_org_search_results(
+                location_params, org_filter_params, donation_request_filter_params, offset
+            )
+            response = _filter_org_list_by_donation_requests(org_search_results, item_type)
+            return Response(response)
+        except Exception as e:
+            logging.error(e)
+            return Response(status=status.HTTP_400_BAD_REQUEST)
+
+
+def _process_params(query_params):
+    is_dropoff = query_params.get('is_dropoff')
+    is_pickup = query_params.get('is_pickup')
+    is_mail = query_params.get('is_mail')
+    org_type = query_params.get('org_type')
+    item_type = query_params.get('item_type')
+    lat = query_params.get('lat')
+    lon = query_params.get('lon')
+    radius = query_params.get('radius', 10)
+    zipcode = query_params.get('zipcode')
+
+    location_params = {
+        'zipcode': zipcode,
+        'lat': float(lat) if lat else None,
+        'lon': float(lon) if lon else None,
+        'radius': radius,
+    }
+    org_filter_params = {}
+    if org_type:
```

Comment:
Just want to make sure this has been tested, i.e if `org_type` isn't in `query_params`, no error is thrown right? It's just an empty object.

Reply:
yup!